### PR TITLE
Remove EquationUtil

### PR DESCRIPTION
**Pull request by geofjamg**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
feature

**What is the current behavior?** *(You can also link to an open issue here)*
The `EquationUtil` allows to save and restore equations state change because of a contingency.

**What is the new behavior (if this is a feature change)?**
We don't use `EquationUtil` anymore but only relies on bus/branch disabling. It means that there is no more direct access to equation system from LF/AS/Sensi. We always interact with the `LfNetwork` which automatically  reflects changes to the equation system. 

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/sensi/AcSensitivityAnalysis.java [hunk 1]
```diff
@@ -112,6 +109,9 @@ private void calculatePostContingencySensitivityValues(List<LfSensitivityFactor<
                                                            LoadFlowParameters lfParameters, OpenLoadFlowParameters lfParametersExt,
                                                            String contingencyId, int contingencyIndex, SensitivityValueWriter valueWriter,
                                                            Reporter reporter, boolean hasTransformerBusTargetVoltage) {
+        for (LfBranch branch : lfContingency.getBranches()) {
```
Comment (geofjamg) [inline src/main/java/com/powsybl/openloadflow/sensi/AcSensitivityAnalysis.java:112]: @annetill we can safely replace by LfContingency.apply ?

Comment (annetill) [reply src/main/java/com/powsybl/openloadflow/sensi/AcSensitivityAnalysis.java:112]: It is on my TODO list, I think it should work but I need to add some tests and to perform real tests. My question is: in an AC sensitivity analysis, do you need to support transformers control such as voltage or active power control ? Do we need to support shunt voltage control ?

Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/sensi/AcSensitivityAnalysis.java:114]: For a sensitivity analysis, everything regarding control does not make really sense.

File: src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystemUpdater.java [hunk 2]
```diff
@@ -84,22 +44,48 @@ public void onShuntVoltageControlChange(LfShunt controllerShunt, boolean newVolt
...
-                bus.getVoltageControl().ifPresent(this::updateVoltageControl);
+                equationSystem.getEquation(bus.getNum(), AcEquationType.BUS_TARGET_PHI)
+                        .ifPresent(eq -> eq.setActive(!bus.isDisabled()));
+                equationSystem.getEquation(bus.getNum(), AcEquationType.BUS_TARGET_P)
+                        .ifPresent(eq -> eq.setActive(!bus.isDisabled() && !bus.isSlack()));
+                // set voltage target equation inactive, various voltage control will set next to the correct value
+                equationSystem.getEquation(bus.getNum(), AcEquationType.BUS_TARGET_V)
+                        .orElseThrow()
+                        .setActive(false);
+                bus.getVoltageControl().ifPresent(voltageControl -> AcEquationSystem.updateGeneratorVoltageControl(voltageControl, equationSystem));
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystemUpdater.java:80]: Do you miss `controllerBus.getReactivePowerControl().ifPresent` here?

Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystemUpdater.java:80]: Yes, fixed

**Review by annetill — APPROVED** [review]

**Pull request merged**